src/components/AutoCompleteTags/AutoCompleteTags.js

```diff
@@ -13,74 +13,95 @@
  * limitations under the License.
  */
 
-import React, { Fragment, Component } from 'react';
+import React, { Fragment, useState } from 'react';
 import PropTypes from 'prop-types';
 import styled from '@emotion/styled';
-import { remove, includes, isEmpty } from 'lodash/fp';
+import { css } from '@emotion/core';
+import { isEmpty, map, filter, difference } from 'lodash/fp';
 
 import AutoCompleteInput from '../AutoCompleteInput';
 import Tag from '../Tag';
 
-const TagsWrapper = styled('div')`
-  margin-top: ${props => props.theme.spacings.kilo};
-  /* this *hack* is to not allow the tags to be visible below the overlay */
-  padding: 0 1px;
+const TagsWrapper = styled('div')(
+  ({ theme }) => css`
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    margin-top: ${theme.spacings.kilo};
+  `
+);
 
-  span {
-    display: inline-block;
-    margin-bottom: ${props => props.theme.spacings.byte};
-  }
-`;
-
-class AutoCompleteTags extends Component {
-  static propTypes = {
-    onChange: PropTypes.func.isRequired,
-    availableTags: PropTypes.arrayOf(PropTypes.string).isRequired
-  };
-
-  state = { tags: [] };
-
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.tags.length !== this.state.tags.length) {
-      this.props.onChange(this.state.tags);
+const StyledTag = styled(Tag)(
+  ({ theme }) => css`
+    margin-top: ${theme.spacings.byte};
+    &:first-of-type: {
+      margin-top: 0;
     }
-  }
+  `
+);
 
-  handleAddTag = option =>
-    this.setState(({ tags }) => ({ tags: [...tags, option] }));
+const AutoCompleteTags = ({
+  availableTags,
+  selectedTags,
+  onChange,
+  ...inputProps
+}) => {
+  const [selected, setSelected] = useState(selectedTags);
```

Comment:
Head's up! Circuit UI v1 uses an outdated version of React that does not support hooks. If you intend to merge and use this before v2 is released, you'll need to remove any usage of hooks.